server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1359,7 +1359,9 @@ private static void validateCompositeTemplate(
                 // shard id and the current timestamp
                 xContentRegistry,
                 tempIndexService.newSearchExecutionContext(0, 0, null, () -> 0L, null, emptyMap()),
-                tempIndexService.dateMathExpressionResolverAt(System.currentTimeMillis())
+                tempIndexService.dateMathExpressionResolverAt(System.currentTimeMillis()),
+                // TODO: any way to get system names here?
+                alias -> false
```

Comment:
Note to reviewer: does this need to be fixed? It seems like if we don't catch system names in the validation stage, we may end up catching them elsewhere. If we do need to fix this, can we do it in a follow-up?